src/compiler/crystal/macros/methods.cr

```diff
@@ -644,7 +644,13 @@ module Crystal
           StringLiteral.new(@value + piece)
         end
       when "camelcase"
-        interpret_argless_method(method, args) { StringLiteral.new(@value.camelcase) }
+        interpret_argless_method(method, args) do
+          lower = ((nargs = named_args) && (lower_arg = nargs["lower"]?)) ? lower_arg : BoolLiteral.new(false)
+
+          raise "named argument 'lower' to StringLiteral#camelcase must be a bool, not #{lower.class_desc}" unless lower.is_a?(BoolLiteral)
```

Comment:
?

```suggestion
          raise "named argument 'lower' to StringLiteral#camelcase must be a BoolLiteral, not #{lower.class_desc}" unless lower.is_a?(BoolLiteral)
```

Reply:
All the other exceptions like this use the non `*Literal` style.  `must be a number`, `must be a string`, etc.

Reply:
I see. Lets leave it then. It just feels a bit unconnected because when you provide say a string, the error message would say `... must be a bool, not StringLiteral`.

Reply:
True, possibly could be another issue to determine how we want to display the error and to update all the current usages.

Reply:
True, possibly could be another issue to determine how we want to display the error and to update all the current usages.